pkg/tsdb/azuremonitor/azure-log-analytics-datasource_test.go

```diff
@@ -0,0 +1,378 @@
+package azuremonitor
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/grafana/grafana/pkg/components/null"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/tsdb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildingAzureLogAnalyticsQueries(t *testing.T) {
```

Comment:
<3

Reply:
It's really nice - like it much more than the convey bdd approach.